### PR TITLE
[PW-7186] Create REST endpoints for Order API implementation

### PR DESCRIPTION
**Pull request by RokPopov**
## Summary
This PR creates API endpoints to check giftcard balance, create order and cancel order through order API for the gift cards partial payment implementation.

## Tested scenarios
Made the following calls with Postman and made sure the responses were successful:
- `/store-api/adyen/payment-methods/balance`
- `/store-api/adyen/orders`
- `/store-api/adyen/orders/cancel`

### CODE REVIEW

File: src/Controller/StoreApiController.php [hunk 1]
```diff
@@ -32,9 +32,12 @@
 use Adyen\Shopware\Service\ConfigurationService;
```
Comment (peterojo) [inline src/Controller/StoreApiController.php:32]: Maybe it's time to consider splitting this up into different controllers

Comment (RokPopov) [reply src/Controller/StoreApiController.php:32]: Thanks @peterojo,  created a separate ticket about it

**Review by peterojo — APPROVED** [review]

File: src/Controller/StoreApiController.php [hunk 2]
```diff
@@ -185,6 +209,63 @@ public function getPaymentMethods(SalesChannelContext $context): JsonResponse
...
+     *     name="store-api.action.adyen.orders",
+     *     methods={"POST"}
+     * )
+     *
+     * @param SalesChannelContext $context
+     * @return JsonResponse
+     */
+    public function getOrders(SalesChannelContext $context, Request $request): JsonResponse
+    {
+        $orderId = $request->request->get('orderId');
```
Comment (peterojo) [inline src/Controller/StoreApiController.php:246]: At this point we won't yet have a shopware order ID, this should actually be a create orders endpoint

**Review by peterojo — CHANGES_REQUESTED** [review]

File: src/Controller/StoreApiController.php [hunk 3]
```diff
@@ -185,6 +209,63 @@ public function getPaymentMethods(SalesChannelContext $context): JsonResponse
...
+     * @Route(
+     *     "/store-api/adyen/orders",
+     *     name="store-api.action.adyen.orders",
+     *     methods={"POST"}
+     * )
+     *
+     * @param SalesChannelContext $context
+     * @return JsonResponse
+     */
+    public function getOrders(SalesChannelContext $context, Request $request): JsonResponse
```
Comment (peterojo) [inline src/Controller/StoreApiController.php:244]: could you rename this here and also in the service class to `createOrder`?

File: src/Controller/StoreApiController.php [hunk 4]
```diff
@@ -185,6 +209,63 @@ public function getPaymentMethods(SalesChannelContext $context): JsonResponse
...
+     * @Route(
+     *     "/store-api/adyen/orders/cancel",
+     *     name="store-api.action.adyen.orders.cancel",
+     *     methods={"POST"}
+     * )
+     *
+     * @param SalesChannelContext $context
+     * @return JsonResponse
+     */
+    public function getOrdersCancel(SalesChannelContext $context, Request $request): JsonResponse
```
Comment (peterojo) [inline src/Controller/StoreApiController.php:261]: also this should be a request to `cancelOrder`

Comment (peterojo) [reply src/Controller/StoreApiController.php:261]: naming these functions as getters would be a bit confusing

**Review by peterojo — APPROVED** [review]

File: src/Service/PaymentMethodsBalanceService.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    public function __construct(
+        ConfigurationService $configurationService,
+        ClientService $clientService,
+        LoggerInterface $logger
+    ) {
+        $this->configurationService = $configurationService;
+        $this->clientService = $clientService;
+        $this->logger = $logger;
+    }
+
```
Comment (candemiralp) [inline src/Service/PaymentMethodsBalanceService.php:55]:
```suggestion
```

**Review by candemiralp — APPROVED** [review]

**Review by peterojo — APPROVED** [review]

**Pull request merged**